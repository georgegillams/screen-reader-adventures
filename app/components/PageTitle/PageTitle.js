import React from 'react';
import PropTypes from 'prop-types';
import { PageTitle } from 'gg-components/Typography';

const PageTitleWrapper = props => {
  const { link, ...rest } = props;
  const hasLink = !!link;

  return (
    <PageTitle
      link={link}
      headingProps={hasLink ? {} : { id: 'autoFocus' }}
      linkProps={hasLink ? { id: 'autoFocus' } : {}}
      {...rest}
    />
  );
};

PageTitleWrapper.propTypes = {
  ticketTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default PageTitleWrapper;
