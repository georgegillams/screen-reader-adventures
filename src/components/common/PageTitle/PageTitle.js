import React from 'react';
import PropTypes from 'prop-types';
import { PageTitle as GGPageTitle } from 'gg-components/PageTitle';
import TextLink from 'components/common/TextLink';
import Head from 'next/head';
import appConfig from 'helpers/appConfig';

const PageTitle = props => {
  const { link, name, pageTitle, ...rest } = props;

  let result = null;
  if (!link) {
    result = <GGPageTitle name={name} {...rest} />;
  } else {
    result = (
      <GGPageTitle
        name={name}
        link={link}
        renderLink={(linkHref, linkName, linkClassName) => {
          return (
            <TextLink href={linkHref} className={linkClassName}>
              {linkName}
            </TextLink>
          );
        }}
        {...rest}
      />
    );
  }

  const pageTitleFinal = `${pageTitle || name} - ${appConfig.projectTitle}`;

  return (
    <>
      <Head>
        <title>{pageTitleFinal}</title>
        <meta key="og:title" name="og:title" content={pageTitleFinal} />
      </Head>
      {result}
    </>
  );
};

PageTitle.propTypes = {
  link: PropTypes.object,
  name: PropTypes.string.isRequired,
  pageTitle: PropTypes.string,
};

PageTitle.defaultProps = {
  link: null,
  pageTitle: null,
};

export default PageTitle;
