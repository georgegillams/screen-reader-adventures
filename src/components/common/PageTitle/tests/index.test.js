import React from 'react';

import { render } from '@testing-library/react';

import PageTitle from '..';

describe('<PageTitle />', () => {
  const children = 'some text';

  it('Should render correctly', () => {
    const { container } = render(<PageTitle name="Test">{children}</PageTitle>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(
      <PageTitle name="Test" className={'test-custom-classname'}>
        {children}
      </PageTitle>
    );

    expect(container).toMatchSnapshot();
  });
  // This test is failing - revisit
  it('Should render correctly with other props', () => {
    const { container } = render(
      <PageTitle name="Test" pageTitle="Different" link={{ to: '/test', text: 'test' }}>
        {children}
      </PageTitle>
    );

    expect(container).toMatchSnapshot();
  });
});
