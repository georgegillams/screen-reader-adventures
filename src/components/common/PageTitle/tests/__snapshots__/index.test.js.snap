// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> Should render correctly 1`] = `
<div>
  <h1
    class="page-title__heading"
  >
    Test
  </h1>
  some text
</div>
`;

exports[`<PageTitle /> Should render correctly with custom className 1`] = `
<div>
  <h1
    class="page-title__heading test-custom-classname"
  >
    Test
  </h1>
  some text
</div>
`;

exports[`<PageTitle /> Should render correctly with other props 1`] = `
<div>
  <a
    class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
    href="/test"
  >
    â‡  test
  </a>
  <h1
    class="page-title__heading page-title__heading--with-link"
  >
    Test
  </h1>
  some text
</div>
`;
